### ISSUE TITLE
WebRTC communication problem

### ISSUE THREAD

**Issue by StevenMichael3213** [tdlib/td#1736]
After I used the latest tb code to recompile the 1.7.3 version of the td library, and applied it to Telegram React, I found some incompatibility due to API changes. After I updated, I dealt with some incompatibility. However, after the update, after the user spoke on the Telegram React end, the voice could not be transmitted normally to the other end (Telegram Desktop). I checked the SDP log and found that it was basically the same as the log before I updated the td library. Below is a screenshot.

I’m not sure if the API about the communication part of WebRTC has been modified in the td library? Or when I call joinGroupCall, are the parameters in the payload missing or wrong?
Currently my payload is in the string form of {ufrag, pwd, hash, setup, fingerprint, ssrc}.

![Image](https://user-images.githubusercontent.com/31765462/141657140-03ed54a1-918d-4202-af9b-998733e1f674.png)
![Image](https://user-images.githubusercontent.com/31765462/141657142-b5ae27c9-43f9-43db-a733-b99e12ee491a.png)
![Image](https://user-images.githubusercontent.com/31765462/141657144-a272d832-e321-4b15-ab71-b41d2afaeb95.png)
![Image](https://user-images.githubusercontent.com/31765462/141657146-ec662a10-cd15-4e47-ba9f-5f61b8ac0742.png)

**Comment by levlam** [comment]
Current stable Telegram Descktop version can have issues with calls from browsers. At the time the calls were supported in Telegram React, only Telegram iOS and other browsers were able to accept them. Telegram Desktop is supposed to support calls with browsers only after the next stable release.

**Comment by StevenMichael3213** [comment]
After I used the latest version of the code, I also found another problem.
After I joined the call, I couldn’t receive the events updated by other participants in the call. That is, in 'updateGroupCallParticipant', I can only receive my own status, such as mute, unmute, and other people’s status.  I can't see it at all.

But about 12 hours ago, I was still able to receive other people's status. 11 hours ago, when I posted the last issue, I couldn't receive it either.

Excuse me, is this caused by the updated code on the server side?

![Image](https://user-images.githubusercontent.com/31765462/141671256-73bc9410-4832-4cac-9cf1-854b036f0bea.png)

**Comment by levlam** [comment]
Have you loaded full list of group call participants using `loadGroupCallParticipants`. There is a flag `groupCall.loaded_all_participants` to denote this.

**Comment by StevenMichael3213** [comment]
Uh, yes. I changed the original code. In the original code, loadGroupCallParticipants was called twice at the same time. I think it was repeated the second time, so I commented out the second call. As a result, there was a problem. Thanks. After restoring the previous code, I can receive the status of other people.

**Comment by StevenMichael3213** [comment]
The above is an episode.

Going back to the original question, I tested the Android end and the IOS end. After they joined the call, they couldn't receive the voice from the Telegram React end.

In addition, after sending the `setGroupCallParticipantIsSpeaking` event on the Telegram React side, neither the Desktop nor the Android side can receive the status of `is_speaking` as true.
The following is a screenshot of the setGroupCallParticipantIsSpeaking event sent by the Telegram React side.
![Image](https://user-images.githubusercontent.com/31765462/141674166-e3397941-1d7b-4112-8425-0414adc51cd6.png)

Is this caused by the inability of the Desktop side to accept the voice from the Telegram React side?

On the contrary, if I speak on the Desktop side, I cannot receive the change of `is_speaking` state on the Telegram React side(That is, the notification of updateGroupCallParticipant cannot be received).
This is strange. What are the possible reasons?

**Comment by levlam** [comment]
Calls with web versions weren't tested after initial implementation and there were really a lot of changes in tgcalls after that, so there can be a lot of inconsistencies. I would expect that the calls will actually work in browsers only after they are implemented in a new web version `webk` or `webz`.

**Comment by StevenMichael3213** [comment]
It's great. I am looking forward to calls to work normally on the browser.

I looked at the code of the td library, and I found such a piece of code. Through this code, I finally knew how to assign a value to the payload in` JoinGroupCall`.

But I have a question. In the code that assigns the parameter payload, the parameters such as `urfag`, `pwd`, `hash`, `setup`, and `fingerprint` are all hard-coded. They are not the parameters passed by the acquired user to `joinGroupCall`. Is this code free of bugs?
![image](https://user-images.githubusercontent.com/31765462/141740433-2b9dbe08-eb13-4da8-9d02-904a266d723c.png)
